networks/osmosis-1/upgrades/v7/guide.md

```diff
@@ -0,0 +1,105 @@
+# v6 to v7 Upgrade Guide
+
+
+## Memory Requirements
+
+For this upgrade, nodes will need a total of 64GB of memory. This must consist of a **minimum** of 32GB of RAM, while the remaining 32GB can be swap. For best results, use 64GB of physical memory.
+
+
+Short version swap setup instructions:
+```sh
+sudo swapoff -a
+sudo fallocate -l 32G /swapfile
+sudo chmod 600 /swapfile
+sudo mkswap /swapfile
+sudo swapon /swapfile
+```
+
+To persist swap after restart:
+```sh
+sudo cp /etc/fstab /etc/fstab.bak
+echo '/swapfile none swap sw 0 0
+```
+
+In depth swap setup instructions: https://www.digitalocean.com/community/tutorials/how-to-add-swap-space-on-ubuntu-20-04
+
+
+## Install and setup Cosmovisor
+
+We highly recommend validators use cosmovisor to run their nodes. This will make low-downtime upgrades smoother, as validators don't have to manually upgrade binaries during the upgrade, and instead can pre-install new binaries, and cosmovisor will automatically update them based on on-chain SoftwareUpgrade proposals.
+
+You should review the docs for cosmovisor located here: https://docs.cosmos.network/master/run-node/cosmovisor.html
+
+If you choose to use cosmovisor, please continue with these instructions:
+
+To install Cosmovisor:
+
+```sh
+cd $HOME
+git clone https://github.com/cosmos/cosmos-sdk
+cd cosmos-sdk
+git checkout v0.44.0
+make cosmovisor
+cp cosmovisor/cosmovisor $GOPATH/bin/cosmovisor
+cd $HOME
```

Comment:
I think we can now do `go install github.com/cosmos/cosmos-sdk/cosmovisor/cmd/cosmovisor@v1.0.0` right?

Reply:
You are correct, just tested it out and it worked (got an error log when starting cosmovisor `failed to read error="lstat /root/.osmosisd/cosmovisor/current/upgrade-info.json: no such file or directory"`, but it still works. I think this because I didn't fill out the "info" portion of the software upgrade prop on testnet?)

Reply:
Yeah I think thats why you get that error message as well.

We should test with something in the field, I believe theres some JSON structure we have to adhere to.